tensorrt_llm/_torch/cute_dsl_kernels/blackwell/low_m_bf16_direct.py

```
@@ -0,0 +1,429 @@
+        raise ValueError(f"unsupported block_size={tactic.block_size}")
+    if tactic.outputs_per_block not in _SUPPORTED_OUTPUTS_PER_BLOCK:
+        raise ValueError(f"unsupported outputs_per_block={tactic.outputs_per_block}")
+    if not 1 <= m <= MAX_M:
+        raise ValueError(f"direct GEMM requires 1 <= M <= {MAX_M}, got {m}")
+    if not 1 <= tactic.rows_per_block <= m or m % tactic.rows_per_block:
+        raise ValueError(f"rows_per_block={tactic.rows_per_block} must divide M={m}")
+    if n <= 0 or n % tactic.outputs_per_block:
+        raise ValueError(f"N={n} must be divisible by outputs_per_block={tactic.outputs_per_block}")
+    k_tile = tactic.block_size * _VECTOR_WIDTH
+    if k <= 0 or k % k_tile:
+        raise ValueError(f"K={k} must be divisible by block_size×{_VECTOR_WIDTH}={k_tile}")
```

Ruff (0.16.1)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)